sec2_the_ruby_programming_language/2.3_number.rb

```diff
@@ -0,0 +1,75 @@
+# Numbers
+puts 1 + 2       #3
+puts 2 - 1       #1
+puts 10 * 2      #20
+puts 10 / 4      #2
+puts 10.0 / 4    #2.5
+puts 10 / 4.0    #2.5
+puts 10 / 4.to_f #2.5
+puts (10/4).to_f #2.0
```

Comment:
Why?

Reply:
This is considered as the usual arithmetic conversions, when either type is the float number, the end result will also become a float number.

Reply:
No idea how does that answer the question.

Reply:
This is because all the answers are converted to become a float number.

Reply:
Still have no idea what are you talking about.
Are you sure you know what exactly I am asking about?

Reply:
I'm not sure about the question. May I have more clarification?

Reply:
My question: 

```
(10/4).to_f
```

Think in terms of mathematics:
10 divide by 4 is 2.5.

so:
(10/4) = (2.5)
(2.5).to_f = 2.5

Why is it 2.0 instead of 2.5?

Reply:
This is because the computation will be carried out according to the precedence level of the operator. Based on the picture here, the parentheses have higher precedence over another operator.

In this case, the `10/4` is being encapsulated in the parenthesis in which it will be computed first then only will be converted into a float number by `.to_f`

The sequence of the computation shall be looked like as below:
1. `(10/4).to_f`
2. `(2).to_f`
3. `2.0` (final answer)

![image](https://user-images.githubusercontent.com/72864300/163666953-2be7969e-0375-4b34-8929-d668e6eb9077.png)